llama_index/vector_stores/google/generativeai/base.py

```diff
@@ -0,0 +1,231 @@
+
+    def add(self, nodes: List[BaseNode]) -> List[str]:
+        try:
+            import google.ai.generativelanguage as genai
+
+            import llama_index.vector_stores.google.generativeai.genai_extension as genaix
+        except ImportError:
+            raise ImportError(_import_err_msg)
+
+        _logger.debug(f"\n\nGoogleVectorStore.add(nodes={nodes})")
+
+        client = cast(genai.RetrieverServiceClient, self.client)
+
+        for nodeGroup in _groupNodesBySource(nodes):
+            source = nodeGroup.source_node
+            document_id = source.node_id
+            document = genaix.get_document(
+                corpus_id=self.corpus_id, document_id=document_id, client=client
+            )
+
+            if not document:
+                genaix.create_document(
+                    corpus_id=self.corpus_id,
+                    display_name=source.metadata.get("file_name", None),
+                    document_id=document_id,
+                    metadata=source.metadata,
+                    client=client,
+                )
+
+            genaix.batch_create_chunk(
+                corpus_id=self.corpus_id,
+                document_id=document_id,
+                texts=[node.get_content() for node in nodeGroup.nodes],
+                metadatas=[node.metadata for node in nodeGroup.nodes],
+                client=client,
+            )
+
+        return [node.node_id for node in nodes]
+
+    def delete(self, ref_doc_id: str, **delete_kwargs: Any) -> None:
+        try:
+            import google.ai.generativelanguage as genai
+
+            import llama_index.vector_stores.google.generativeai.genai_extension as genaix
+        except ImportError:
+            raise ImportError(_import_err_msg)
+
+        _logger.debug(f"\n\nGoogleVectorStore.delete(ref_doc_id={ref_doc_id})")
+
+        client = cast(genai.RetrieverServiceClient, self.client)
+        genaix.delete_document(
+            corpus_id=self.corpus_id, document_id=ref_doc_id, client=client
+        )
+
+    def query(self, query: VectorStoreQuery, **kwargs: Any) -> VectorStoreQueryResult:
+        try:
+            import google.ai.generativelanguage as genai
+
+            import llama_index.vector_stores.google.generativeai.genai_extension as genaix
+        except ImportError:
+            raise ImportError(_import_err_msg)
+
+        _logger.debug(f"\n\nGoogleVectorStore.query(query={query})")
+
+        client = cast(genai.RetrieverServiceClient, self.client)
+
+        relevant_chunks: List[genai.RelevantChunk] = []
+        if query.doc_ids is None:
+            relevant_chunks = genaix.query_corpus(
+                corpus_id=self.corpus_id,
+                query=query.query_str or "what?",
+                filter=_convertFilter(query.filters),
+                k=query.similarity_top_k,
+                client=client,
+            )
+        else:
+            for doc_id in query.doc_ids:
+                relevant_chunks.extend(
```

Comment:
Since we're combining chunks from multiple documents, should we sort `relevant_chunks` by relevance score after this for loop so we consistently return in descending order?